### PR TITLE
[PRM-205]: added financial penalty threshold reached content

### PR DESCRIPTION
**Pull request by BackRowOla**

### CODE REVIEW

**Comment by BackRowOla** [comment]
Screenshot 2021-05-06 at 16 13 03

File: app/controllers/IndexController.scala [hunk 1]
```diff
@@ -41,11 +40,13 @@ class IndexController @Inject()(view: IndexView,
   extends FrontendController(controllerComponents) with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
+    logger.debug("[IndexController][onPageLoad]: Hitting the controller, before 'for->yield")
```
Comment (LewisMitchell) [inline app/controllers/IndexController.scala:43]: I don't think we need this logger message anymore.

File: app/controllers/IndexController.scala [hunk 2]
```diff
@@ -41,11 +40,13 @@ class IndexController @Inject()(view: IndexView,
...
     for {
       lSPData <- penaltiesService.getLspDataWithVrn(EnrolmentKeys.constructMTDVATEnrolmentKey(request.vrn))
       contentToDisplayAboveCards = pageHelper.getContentBasedOnPointsFromModel(lSPData)
       summaryCards = cardHelper.populateCard(lSPData.penaltyPoints)
     } yield {
+      logger.debug("[IndexController][onPageLoad]: Hitting the controller, about to call view")
```
Comment (LewisMitchell) [inline app/controllers/IndexController.scala:49]: I don't think we need this logger message anymore.

File: app/viewmodels/IndexPageHelper.scala [hunk 3]
```diff
@@ -65,51 +66,60 @@ class IndexPageHelper @Inject()(p: views.html.components.p,
...
+      case (currentPoints, threshold, _) if currentPoints == threshold - 1 => {
         html(
           renderPointsTotal(currentPoints),
           warningText(stringAsHtml(messages("lsp.pointSummary.penaltyPoints.overview.warningText"))),
-          p(getPluralOrSingularContentForOverview(currentPoints, etmpData.lateSubmissions))
+          p(getPluralOrSingularContentForOverview(currentPoints, etmpData.lateSubmissions)),
+          getGuidanceLink
         )
       }
       //TODO: replace this with max scenarios - added this so we don't get match errors.
```
Comment (LewisMitchell) [inline app/viewmodels/IndexPageHelper.scala:100]: Probably can remove this TODO now.

File: app/viewmodels/IndexPageHelper.scala [hunk 4]
```diff
@@ -65,51 +66,60 @@ class IndexPageHelper @Inject()(p: views.html.components.p,
...
-      case _ => {
-        p(content = html(stringAsHtml("")))
+      case (currentPoints, threshold, _) if currentPoints >= threshold => {
```
Comment (LewisMitchell) [inline app/viewmodels/IndexPageHelper.scala:101]: Checked the precedence - I think this case statement should be put below the first case statement.

File: app/viewmodels/IndexPageHelper.scala [hunk 5]
```diff
@@ -124,4 +134,13 @@ class IndexPageHelper @Inject()(p: views.html.components.p,
       )
     })
   }
+
+  def getGuidanceLink(implicit messages: Messages): HtmlFormat.Appendable = p(
```
Comment (LewisMitchell) [inline app/viewmodels/IndexPageHelper.scala:138]: Nice 🚀

File: test/views/IndexViewSpec.scala [hunk 6]
```diff
@@ -142,13 +141,6 @@ class IndexViewSpec extends SpecBase with ViewBehaviours {
           doc.select(Selectors.breadcrumbWithLink(1)).attr("href") shouldBe appConfig.vatOverviewUrl
         }
 
-        "have a link to external guidance which opens in a new tab" in {
```
Comment (LewisMitchell) [inline test/views/IndexViewSpec.scala:145]: Do we not want to keep this in?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
LGTM. Just a few minor things.

File: test/views/IndexViewSpec.scala (hunk 6 above)
Comment (BackRowOla) [reply test/views/IndexViewSpec.scala:145]: The test can't find the selector as the content has been moved out of the index view so I assumed it wasn't needed in here and moved it to the helper.

Comment (LewisMitchell) [reply test/views/IndexViewSpec.scala:145]: That makes sense - just thought that if someone were to remove the helper from the view in the controller then the content would be missing. Happy to leave it out.

File: conf/messages [hunk 7]
```diff
@@ -34,6 +34,10 @@ lsp.pointSummary.penaltyPoints.overview.whatHappensWhenThresholdExceeded = If yo
 
 lsp.pointSummary.penaltyPoints.overview.warningText = You’ll get a £200 penalty if you submit another VAT Return late.
 
+lsp.onThreshold.p1 = You have reached the financial penalty threshold.
+lsp.onThreshold.p2 = Until you bring your account up to date, you’ll have to pay a £200 penalty every time you submit a VAT Return late.
+lsp.onThreshold.link = Show me how to bring this account up to date
```
Comment (LewisMitchell) [inline conf/messages:39]: Sorry - didn't notice this before. Could you copy this over to messages.cy?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
LGTM. I missed one thing, sorry!

File: conf/messages (hunk 7 above)
Comment (LewisMitchell) [reply conf/messages:39]: The lsp.onThreshold.link message

**Review by LewisMitchell — APPROVED** [review]
All good, waiting for tests.

**Pull request merged**